### ISSUE TITLE
Polymer (demo + website made with Dart) loads extremely slow on IE11

### ISSUE THREAD

**Issue by brnjamin** [Polymer/polymer#891]
http://stackoverflow.com/questions/26768081/website-application-made-with-polymer-components-loads-extremely-slow-on-ie11 

We are considering polymer technology (& the dart language) to develop a public/internet web application / web site. When testing /validating the approach, we encountered a potential show-stopper. The (basic) site is performing rather o.k. on different browsers & platforms, but it is extremely slow on Internet Explorer 11 on Windows 8.1.

The problem does not seem to be related to our specific code, as it seems easy to reproduce the behaviour using, for example, the polymer demo (menu button): https://www.polymer-project.org/components/paper-elements/demo.html#paper-menu-button

On IE11 (11.0.9600.17351), it takes more than 16 seconds to load that page. With the “UI Responsiveness” profiler of IE11, a .diagsession file was created: http://novonov.be/tmp/polymer-paper-menu-button-demo.diagsession At first sight, the problem does not seem to be related to network traffic / number of files to be downloaded. 15.73 seconds are spent in “DOM event (readystatechanged)”. We did not test another version of IE - only IE11.

**Comment by Meligy** [comment]
What if you define your styles inside `<core-style>` not CSS files?

**Comment by jakemac53** [comment]
That should work fine as well

**Comment by Meligy** [comment]
You mean adding `no-shim` to `core-style` calls?

**Comment by jakemac53** [comment]
Oh, sorry I didn't quite read what you said correctly. I don't know if you can no-shim the contents of a core-style, but by using core-style you can at least make sure it only shims it once. The big issue here was it was shimming it multiple times because the stylesheet was being included in multiple places.

**Comment by shogun70** [comment]
FYI I have found a performance regression in IE11 / Edge that applies to non-empty custom-elements when they are not actually in a document - they might be contained in a DocumentFragment or just part of a tree where the root-element has no parentNode. It *seems* unknown elements trigger some relayout code even when they aren't in a document. 

I'm not sure if this regression applies to the current issue or is significant to any other part of Polymer but I wouldn't be surprised. Certainly IE has no shortage of surprises. 

I submitted the details to MSFT at:
  https://connect.microsoft.com/IE/feedback/details/1776195/ie11-edge-performance-regression-with-dom-fragments
